From 44f3a9539487724732abf8246b3c6b45fe6c09a9 Mon Sep 17 00:00:00 2001
From: David Holsgrove <david.holsgrove@petalogix.com>
Date: Tue, 27 Mar 2012 22:32:54 +1000
Subject: [PATCH 25/79] microblaze: Remove MASK_REORDER from default target_flags

and instead only add to target_flags if microblaze v8.30a or greater

Signed-off-by: David Holsgrove <david.holsgrove@petalogix.com>
---
 gcc/config/microblaze/microblaze.c |    5 +++++
 gcc/config/microblaze/microblaze.h |    4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gcc/config/microblaze/microblaze.c b/gcc/config/microblaze/microblaze.c
index c1caaf3..cdea221 100755
--- a/gcc/config/microblaze/microblaze.c
+++ b/gcc/config/microblaze/microblaze.c
@@ -1422,6 +1422,11 @@ microblaze_option_override (void)
           warning (0,
                  "-mxl-reorder can be used only with -mcpu=v8.30.a or greater");
     }
+  else
+   {
+       /* Add to Default target_flags if v8.30a or greater  */
+       target_flags |= MASK_REORDER;
+   }
   if (TARGET_MULTIPLY_HIGH && TARGET_SOFT_MUL)
     error ("-mxl-multiply-high requires -mno-xl-soft-mul");
 
diff --git a/gcc/config/microblaze/microblaze.h b/gcc/config/microblaze/microblaze.h
index 350b227..b1e0249 100755
--- a/gcc/config/microblaze/microblaze.h
+++ b/gcc/config/microblaze/microblaze.h
@@ -58,13 +58,13 @@ extern enum pipeline_type microblaze_pipe;
 
 /* Default target_flags if no switches are specified  */
 #define TARGET_DEFAULT      (MASK_SOFT_MUL | MASK_SOFT_DIV | MASK_SOFT_FLOAT \
-                             | TARGET_ENDIAN_DEFAULT |MASK_REORDER)
+                             | TARGET_ENDIAN_DEFAULT )
 
 /* Do we have CLZ?  */
 #define TARGET_HAS_CLZ      (TARGET_PATTERN_COMPARE && microblaze_has_clz)
 
 /* Do we have SWAPB and SWAPH?  */
-#define TARGET_HAS_SWAP     (TARGET_PATTERN_COMPARE && microblaze_has_swap)
+#define TARGET_HAS_SWAP     (TARGET_REORDER && microblaze_has_swap)
 
 /* The default is to not support PIC.  */
 #define TARGET_SUPPORTS_PIC 0
-- 
1.7.1

